torch/_inductor/kernel/mm.py

```diff
@@ -183,6 +183,11 @@ def tuned_mm(mat1, mat2, *, layout=None):
         log.warning("No choices for GEMM, using ATen backend as fallback")
         return aten_mm.bind((mat1, mat2), aten_layout).output_node()
 
+    from ..external_callable import external_matmul
+
+    for k in external_matmul:
+        choices.append(ExternKernelChoice(k).bind((mat1, mat2), layout))
```

Comment:
We should only construct `ExternKernelChoice` once, since I believe the constructor here registers a new name.  Maybe we need a `functools.lru_cache(None)` to dedupe these.

Reply:
Hi Jason, I see that ExternKernelChoice registers a new name for each new function. And we could run into duplication problem. But could you elaborate more how to use `functools.lrucache(None)` to dedupe these new name?

Reply:
```
@funtools.lru_cache(None)
def lazy_register_extern_choice(fn):
   return ExternKernelChoice(fn)
```
will cause it to only get constructed once.

Reply:
Added deduplication class